LibCore: MimeType for text/plain has base names with lines folded

Most likely a very minor issue and maybe only really impacting Serenity OS, but was highlighted when I used static checker cppcheck 2.13.0 on ladybird code base.  I was checking the Libraries at the time and gave this output


```
LibCore/MimeData.cpp:64:15: error: There is an unknown macro here somewhere. Configuration is required. If sv is a macro then please configure it. [unknownMacro]
    ".shellrc"sv
              ^
```

Seems cppcheck is confused by the sv StringView suffix.  This kind of cppcheck "error" even blocks cppcheck from doing deeper verifications.  Anyway...

MimeType.cpp looks like this:

```
60     ".txt"sv,
61
62     // Base names
63     ".history"sv,
64     ".shellrc"sv
65     "CMakeLists.txt"sv,
66 };
```

Looked at line 64 and realized that a comma was missing at line end, causing the ".shellrc" and "CMakeLists.txt" strings to be folded together as ".shellrcCMakeLists.txt". It should be...

`64     ".shellrc"sv,`

with the comma at end of line.


I ran some manual file:// tests with ladybird, with and without the fix and noticed that CMakeLists.txt was always shown as expected in the browser, but .shellrc would fail to be shown until the fix was in.  So then it was obvious that any file with .txt extension will always work as that extension is already up higher on line 60.  So we can remove the CMakeLists.txt entry from the base names too, as it is unnecessary.

I wrote some tests to check the text/plain MimeType and a few others. Hope the tests are ok.

I have a small commit ready to correct this.

This is my first commit for ladybird! Enjoying getting familiar with ladybird and C++20. Hope this report is ok?!


Regards,

--- a/Libraries/LibCore/MimeData.cpp
+++ b/Libraries/LibCore/MimeData.cpp
@@ -61,8 +61,7 @@ static Array constexpr s_plaintext_suffixes = {
 
     // Base names
     ".history"sv,
-    ".shellrc"sv
-    "CMakeLists.txt"sv,
+    ".shellrc"sv,
 };
 
 // See https://www.iana.org/assignments/media-types/<mime-type> for a list of registered MIME types.

--- a/Tests/LibCore/CMakeLists.txt
+++ b/Tests/LibCore/CMakeLists.txt
@@ -5,6 +5,7 @@ set(TEST_SOURCES
     TestLibCoreFileWatcher.cpp
     # FIXME: Identify and address the commit that caused this to start failing at runtime
     #TestLibCoreMappedFile.cpp
+    TestLibCoreMimeType.cpp
     TestLibCorePromise.cpp
     TestLibCoreSharedSingleProducerCircularQueue.cpp
     # FIXME: Identify and address the commit that caused this to start failing at runtime

--- a/Tests/LibCore/TestLibCoreMimeType.cpp
+++ b/Tests/LibCore/TestLibCoreMimeType.cpp
@@ -1,0 +1,55 @@
+/*
+ * Copyright (c) 2025, the Ladybird developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Vector.h>
+#include <LibCore/MimeData.h>
+#include <LibTest/TestCase.h>
+
+static void check_filename_mimetype(Vector<StringView> const& filepaths, StringView expected_mime_type)
+{
+    for (auto const& filename : filepaths) {
+        dbgln(filename, "\n");
+        auto const& guessed_mime_type = Core::guess_mime_type_based_on_filename(filename);
+        EXPECT_EQ(guessed_mime_type, expected_mime_type);
+    }
+}
+
+auto text_plain_filenames = Vector {
+    "main.c"sv,
+    "hello.txt"sv,
+    ".history"sv,
+    ".shellrc"sv,
+    "CMakeList.txt"sv,
+};
+// FIXME: fails because .xht extension is in MimeType text/html and application/xhtml+xml
+// auto html_filenames = Vector {"about.html"sv, "send-data-blob.htm"sv, "content.xht"sv, "dir/settings.html"sv,};
+auto xhtml_filenames = Vector {
+    "about.xhtml"sv,
+    "content.xht"sv,
+};
+auto gzip_filenames = Vector {
+    "download.iso.gz"sv,
+    "backup.gzip"sv,
+    "hello.html.gz"sv,
+};
+auto markdown_filenames = Vector {
+    "README.md"sv,
+    "changelog.md"sv,
+};
+auto shell_filenames = Vector {
+    "script.sh"sv,
+};
+
+TEST_CASE(various_types_guessed)
+{
+    check_filename_mimetype(text_plain_filenames, "text/plain"sv);
+    // FIXME: fails because .xht extension is in MimeType text/html and application/xhtml+xml
+    // check_filename_mimetype(html_filenames, "text/html"sv);
+    check_filename_mimetype(xhtml_filenames, "application/xhtml+xml"sv);
+    check_filename_mimetype(gzip_filenames, "application/gzip"sv);
+    check_filename_mimetype(markdown_filenames, "text/markdown"sv);
+    check_filename_mimetype(shell_filenames, "text/x-shellscript"sv);
+}

